src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheck.java

```diff
@@ -293,7 +293,7 @@ public final void finishTree(DetailAST rootAST) {
     }
 
     @Override
-    public final void visitToken(DetailAST blockCommentNode) {
+    public void visitToken(DetailAST blockCommentNode) {
```

Comment:
Changing the API of the base class will be a big flag to reviewers. It will go under greater scrutiny.

Javadoc should not need to intercept the java token listener. It's main focus is looking at the javadoc and not things that are not connected to it.

You will need to explain why you need changes like these. You should point out things like this before you go down the wrong path. To me this looks like the wrong path.